assets/app/scripts/directives/events.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .directive('eventIcon', function() {
+    return {
+      restrict: "E",
+      scope: false,
+      template: '<span class="pficon {{type | eventIcon}}" aria-hidden="true" data-toggle="tooltip" data-placement="right"' +
+                'data-original-title="{{type}}"></span>',
+      replace: true,
+      link: function(scope, element, attrs) {
+        scope.type = attrs.type;
+      }
+    };
+  })
+  .directive('events', function($routeParams, $filter, DataService, ProjectsService, Logger) {
+    return {
+      restrict: 'E',
+      scope: {
+        kind: "@",
+      },
+      templateUrl: 'views/directives/events.html',
+      controller: function($scope){
+
+        var filterEvent = function(resourceType, resource, event) {
+          var involvedObjectKind;
+          if (resourceType === "pod") {
```

Comment:
$scope.kind needs to be the actual kind -- "Build", "Pod", "Service", etc. -- then you don't need to chain if / else if blocks. See #6990